source/gridfs/gridfs-spec.md

```diff
@@ -821,11 +847,34 @@ Sets the filename field in the stored file's files collection document to the ne
 Drivers construct and execute an update_one command on the files collection using `{ _id: @id }` as the filter and
 `{ $set : { filename : "new_filename" } }` as the update parameter.
 
-To rename multiple revisions of the same filename, users must retrieve the full list of files collection documents for a
-given filename and execute "rename" on each corresponding `_id`.
+If `renameByName` is not implemented to rename multiple revisions of the same filename, users must retrieve the full
+list of files collection documents for a given filename and execute "rename" on each corresponding `_id`.
 
 If there is no file with the given id, drivers MUST raise an error.
 
+### Renaming stored files by filename
+
+To rename all revisions of a stored file with the specified filename, drivers SHOULD provide the method
+`rename_by_name`:
```

Comment:
One thing I did not think about is if there is already a newer file with the new filename, then it will stay the last revision for this filename.

We can't change the date of the renamed files, as that would change the order of the revisions.

Reply:
Correct that seems like expected behavior to me.